packages/headless/src/controllers/controller/headless-controller.ts

```diff
@@ -35,12 +37,25 @@ export function buildController<T extends object>(
   return {
     subscribe(listener: () => void) {
       listener();
-      prevState = JSON.stringify(this.state);
-      return engine.subscribe(() => {
-        if (hasStateChanged(this.state)) {
-          listener();
+      const symbol = Symbol();
+      let unsubscribe: () => void;
+
+      if (hasNoListeners()) {
```

Comment:
I'm not 100% sure, but there might be an edge case with the current logic, where
- First subscriber registers and initializes `prevState`.
- Second subscriber registers, has a different initial state. This has no effect on `prevState`.
- Second subscriber state changes to match first subscriber's initial state.

No update is triggered?